src/transformers/models/timesformer/modeling_timesformer.py

```diff
@@ -0,0 +1,856 @@
+        self.dropout = nn.Dropout(config.hidden_dropout_prob)
+
+    def forward(self, hidden_states: torch.Tensor) -> torch.Tensor:
+        hidden_states = self.dense(hidden_states)
+        hidden_states = self.dropout(hidden_states)
+
+        return hidden_states
+
+
+# Adapted from https://github.com/facebookresearch/TimeSformer/blob/a5ef29a7b7264baff199a30b3306ac27de901133/timesformer/models/vit.py#L89
+class TimeSformerLayer(nn.Module):
+    def __init__(self, config: TimeSformerConfig, layer_index: int) -> None:
+        super().__init__()
+
+        dim = config.hidden_size
+        attention_type = config.attention_type
+
+        dpr = [
+            x.item() for x in torch.linspace(0, config.drop_path_prob, config.num_hidden_layers)
+        ]  # stochastic depth decay rule
+        drop_path_prob = dpr[layer_index]
+
+        self.drop_path = TimeSformerDropPath(config) if drop_path_prob > 0.0 else nn.Identity()
+        self.attention = TimeSformerAttention(config)
+        self.intermediate = TimeSformerIntermediate(config)
+        self.output = TimeSformerOutput(config)
+        self.layernorm_before = nn.LayerNorm(config.hidden_size, eps=config.layer_norm_eps)
+        self.layernorm_after = nn.LayerNorm(config.hidden_size, eps=config.layer_norm_eps)
+
+        self.config = config
+        self.attention_type = attention_type
+        assert attention_type in ["divided_space_time", "space_only", "joint_space_time"]
+
+        # Temporal Attention Parameters
+        if self.attention_type == "divided_space_time":
+            self.temporal_layernorm = nn.LayerNorm(config.hidden_size, eps=config.layer_norm_eps)
+            self.temporal_attention = TimeSformerAttention(config)
+            self.temporal_dense = nn.Linear(dim, dim)
+
+    def forward(self, hidden_states: torch.Tensor, output_attentions: bool = False):
+        T = self.config.num_frames
+        W = self.config.image_size // self.config.patch_size
+        B = hidden_states.shape[0]
+        num_spatial_tokens = (hidden_states.size(1) - 1) // T
+        H = num_spatial_tokens // W
+
+        if self.attention_type in ["space_only", "joint_space_time"]:
+            self_attention_outputs = self.attention(
+                self.layernorm_before(hidden_states), output_attentions=output_attentions
+            )
+            attention_output = self_attention_outputs[0]
+            outputs = self_attention_outputs[1:]  # add self attentions if we output attention weights
+
+            hidden_states = hidden_states + self.drop_path(attention_output)
+
+            layer_output = self.layernorm_after(hidden_states)
+            layer_output = self.intermediate(layer_output)
+            layer_output = self.output(layer_output)
+            layer_output = hidden_states + self.drop_path(layer_output)
+
+            outputs = (layer_output,) + outputs
+
+            return outputs
+
+        elif self.attention_type == "divided_space_time":
```

Comment:
Same here and below, please update all one-letter variable names to have clearer names

Reply:
fixed 👍